commit db2e367bfc3b119609f837eb973d915f6c550b2f
Author: Yu Watanabe <watanabe.yu+github@gmail.com>
Date:   Wed Apr 24 21:52:13 2019 +0200

    resolve: use bridge or bonding interfaces in degraded-carrier state
    
    Fixes #12285.
    
    (cherry picked from commit aeafd03a2b3d8cd1e5f91d218196578911330f8e)

diff --git a/src/resolve/resolved-link.c b/src/resolve/resolved-link.c
index 44f70aceaa..b35f7baa6c 100644
--- a/src/resolve/resolved-link.c
+++ b/src/resolve/resolved-link.c
@@ -699,7 +699,7 @@ bool link_relevant(Link *l, int family, bool local_multicast) {
                 return false;
 
         (void) sd_network_link_get_operational_state(l->ifindex, &state);
-        if (state && !STR_IN_SET(state, "unknown", "degraded", "routable"))
+        if (state && !STR_IN_SET(state, "unknown", "degraded", "degraded-carrier", "routable"))
                 return false;
 
         LIST_FOREACH(addresses, a, l->addresses)
